### PR TITLE
ProfilePage doPost implementation and it's respective tests

### PR DESCRIPTION
**Pull request by ugonna100**
Added doPost method
Added test for doPost of new UserProfile
Added test for doPost of existing UserProfile
Updated PersistentDataStore as it was previously not working with the UserProfile Interests Map data structure

Added missing names to profile fields in profilepage.jsp and  reindented for readability

Found that i had misnamed getLastTimeOnline method as getlastTimeOnline in different files (some had getLastTimeOnline and some had getlastTimeOnline), this was causing an issue and i've renamed them all to getLastTimeOnline

### CODE REVIEW

File: src/main/java/codeu/controller/ProfilePageServlet.java [hunk 1]
```diff
@@ -71,5 +76,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
+      UUID userId = userStore.getUser(username).getId();
+      UserProfile userProfile = null;
+      String aboutMe = request.getParameter("aboutMe");
+      String profilePicture = request.getParameter("profilePicture");
+      String category = request.getParameter("myFavorites");
+      String subCategory = request.getParameter("subcategory");
+
+      if (userProfileStore.isUserProfileCreated(userId)) {
+        userProfile = userProfileStore.getUserProfile(userId);
+        userProfile.setLastTimeOnline(Instant.now());
```
Comment (ces131) [inline src/main/java/codeu/controller/ProfilePageServlet.java]: About setting the last time online (for future reference I guess), I wonder how often we should update it or how we can go about this. When we do the log out function, we definitely want to update it then

File: src/main/java/codeu/controller/ProfilePageServlet.java [hunk 2]
```diff
@@ -71,5 +76,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
        * Send AboutMe, and Interests data to UserProfile object
```
Comment (kyrabusser) [inline src/main/java/codeu/controller/ProfilePageServlet.java]: You can remove the TODO part of this comment now that it's done. :)

File: src/main/java/codeu/controller/ProfilePageServlet.java (hunk 1 above)
Comment (kyrabusser) [reply src/main/java/codeu/controller/ProfilePageServlet.java]: +1, this is a great thing to think about! It seems to me like setting this when the user logs in or out is a great basic version to get working first.

Comment (ugonna100) [reply src/main/java/codeu/controller/ProfilePageServlet.java]: If we're updating when they login, that should be easy to do, however for when they logout, i think i'd have to figure out a way to track when the user leaves the domain

Most people don't formally log out to sites, they just tend to leave the site in general.

**Comment by ugonna100** [comment]
I can't seem to reply to the review but i forgot to say that i've pulled both commits in and am testing the full flow

**Pull request merged**